packages/react-notion-x/src/third-party/collection-view-calendar.tsx

```diff
@@ -0,0 +1,436 @@
+            )}
+            style={{
+              height: `${
+                collectionView.format?.calendar_properties &&
+                Object.keys(collectionView.format?.calendar_properties).length *
+                  20 +
+                  64
+              }px`
+            }}
+            key={indexI}
+          >
+            {i.dates.map((day, indexY) => (
+              <>
+                <div
+                  className={cs(
+                    'notion-selectable',
+                    'notion-calendar-body-inner-week',
+                    startWeekOnMonday
+                      ? indexY == 5 || indexY == 6
+                        ? 'notion-calendar-body-inner-week-dif'
+                        : ''
+                      : indexY == 0 || indexY == 6
+                      ? 'notion-calendar-body-inner-week-dif'
+                      : ''
+                  )}
+                  key={indexY}
+                >
+                  <div
+                    className={cs(
+                      'notion-calendar-body-inner-day',
+                      day == new Date(Date.now()).getDate() &&
+                        currentYear.getMonth() ==
+                          new Date(Date.now()).getMonth() &&
+                        currentYear.getFullYear() ==
+                          new Date(Date.now()).getFullYear()
+                        ? 'notion-calendar-body-inner-day-today'
+                        : (day == 1 && currentMonth++ == 0) ||
+                          ((day <= 31 || day >= 28) && currentMonth == 1)
+                        ? 'notion-calendar-body-inner-day-this-month'
+                        : 'notion-calendar-body-inner-day-other-month'
+                    )}
+                  >
+                    {day == 1
+                      ? `${
+                          monthsShort[currentYear.getMonth() + currentMonth - 1]
+                        } ${day}`
+                      : day}
+                  </div>
+                </div>
+
+                {blockIds?.map((blockId) => {
+                  const block = recordMap.block[blockId]?.value as PageBlock
+                  if (!block) return null
+
+                  // Get date from calendar view query
+                  const blockDate = getPagePropertyFromId(
+                    collectionView.query2.calendar_by,
+                    block,
+                    recordMap
+                  )
+                  const blockDateDATE = new Date(blockDate as number)
+                  const dayBlock = startWeekOnMonday
+                    ? blockDateDATE.getDay() === 0
+                      ? 6
+                      : blockDateDATE.getDay() - 1
+                    : blockDateDATE.getDay()
+
+                  const titleSchema = collection.schema.title
+                  const titleData = block?.properties?.title
+
+                  if (
+                    new Date(blockDate as number).getDate() == day &&
```

Comment:
why not reuse `blockDateDATE` here instead of recreating the date over and over?